bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,302 @@
+        this.networkTrafficObserver = networkTrafficObserver;
+    }
+
+    private boolean hasAuthenticated() {
+        return authenticated;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializing handler");
+
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+        count = (new Random()).nextInt(65535);
+
+        if (this.socket == null) {
+            this.socket = new RetryableSocket(thingConfig, logger);
+        }
+
+        updateItemStatus();
+
+        if (thingConfig.getPollingInterval() != 0) {
+            refreshHandle = scheduler.scheduleWithFixedDelay(new Runnable() {
+
+                @Override
+                public void run() {
+                    updateItemStatus();
+                }
+            }, 1L, thingConfig.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void dispose() {
+        logger.debug("Disposing of {}", getThing().getLabel());
+        if (refreshHandle != null && !refreshHandle.isDone()) {
+            logger.debug("Cancelling refresh task");
+            boolean cancelled = refreshHandle.cancel(true);
+            logger.debug("Cancellation successful: {}", cancelled);
+        }
+        if (socket != null) {
+            socket.close();
+            socket = null;
+        }
+        super.dispose();
+    }
+
+    private boolean authenticate() {
+        logger.trace("Authenticating");
+        authenticated = false;
+        // When authenticating, we must ALWAYS use the initial values
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+
+        try {
+            byte authRequest[] = buildMessage((byte) 0x65, BroadlinkProtocol.buildAuthenticationPayload(), -1);
+            byte response[] = sendAndReceiveDatagram(authRequest, "authentication");
+            if (response == null) {
+                logger.warn("response from device during authentication was null");
+                return false;
+            }
+            byte decryptResponse[] = decodeDevicePacket(response);
+            this.deviceId = BroadlinkProtocol.getDeviceId(decryptResponse);
+            this.deviceKey = BroadlinkProtocol.getDeviceKey(decryptResponse);
+
+            // Update the properties, so that these values can be seen in the UI:
+            Map<String, String> properties = editProperties();
+            properties.put("id", HexUtils.bytesToHex(deviceId));
+            properties.put("key", HexUtils.bytesToHex(deviceKey));
+            updateProperties(properties);
+            logger.debug("Authenticated with id '{}' and key '{}'", HexUtils.bytesToHex(deviceId),
+                    HexUtils.bytesToHex(deviceKey));
+            authenticated = true;
+            return true;
+        } catch (Exception e) {
+            logger.warn("Authentication failed: ", e);
```

Comment:
```suggestion
            logger.warn("Authentication failed: {}", e.getMessage());
```
Missing the brackets and also if you do not add the e.getMessage() you will get a comment in the report.html file.